### DISCUSSION TITLE
Crypto currency

### DISCUSSION THREAD

**Discussion by Yaarjatt** [community/community#129962] [Programming Help]
### Select Topic Area

Question

### Body

Hi , I am writing a code for finding crypto currency wallets , I uses trial version of some api websites to check balance in generated addresses continuously but after a certain period they banned my accounts , is their any possible way that I can setup my own server to check balance in addresses ??

**Comment by Sworzen1** [comment]
Hello @Yaarjatt 
Please provide more content.
1. Which API did you use ? 
2. Which blockchain ?

**Reply by ebndev** [reply]
Hi @Sworzen1, thanks for taking the time to answer!

@Yaarjatt let us know if you have more questions 😄

**Comment by enzodevs** [comment]
### Hi @Yaarjatt,

Here are some suggestions and questions to help you:

- Which specific cryptocurrencies are you interested in? Different blockchains require different approaches.

- Are you familiar with running a full node? For many cryptocurrencies, this is the most reliable way to query balance information directly from the blockchain.

If you're comfortable with programming, you could:

- Implement your own blockchain parser
- Use libraries like Web3.js for Ethereum-based chains
- Interact directly with RPC nodes

To avoid bans, consider:

- Implementing rate limiting in your code
- Rotating IP addresses
- Distributing requests across multiple services

Remember, while it's possible to set up your own system, it requires significant resources and ongoing maintenance. Make sure the effort aligns with your goals.
